compute/resource_live_table.go

```diff
@@ -0,0 +1,221 @@
+	SparkEnvVars     map[string]string        `json:"spark_env_vars,omitempty"`
+	InstancePoolId   string                   `json:"instance_pool_id,omitempty"`
+}
+
+type PipelinesAutoScale struct {
+	MinWorkers int `json:"min_workers"`
+	MaxWorkers int `json:"max_workers"`
+}
+
+type PipelinesClusterSize struct {
+	NumWorkers int                 `json:"num_workers,omitempty"`
+	Autoscale  *PipelinesAutoScale `json:"autoscale,omitempty"`
+}
+
+type PipelinesNewCluster struct {
+	PipelinesClusterAttributes
+	PipelinesClusterSize
+}
+
+type PipelineCluster struct {
+	Label string `json:"label,omitempty"`
+	PipelinesNewCluster
+}
+
+type PipelinesMavenLibrary struct {
+	Coordinates string   `json:"coordinates"`
+	Repo        string   `json:"repo,omitempty"`
+	Exclusions  []string `json:"exclusions,omitempty"`
+}
+
+type NotebookLibrary struct {
+	Path string `json:"path"`
+}
+
+type PipelineLibrary struct {
+	Jar      string                 `json:"jar,omitempty"`
+	Maven    *PipelinesMavenLibrary `json:"maven,omitempty"`
+	Whl      string                 `json:"whl,omitempty"`
+	Notebook *NotebookLibrary       `json:"notebook,omitempty"`
+}
+
+type Filters struct {
+	Include []string `json:"include,omitempty"`
+	Exclude []string `json:"exclude,omitempty"`
+}
+
+type PipelineSpec struct {
+	Id            string            `json:"id,omitempty"`
+	Name          string            `json:"name,omitempty"`
+	Storage       string            `json:"storage,omitempty"`
+	Configuration map[string]string `json:"configuration,omitempty"`
+	Clusters      []PipelineCluster `json:"clusters,omitempty"`
+	Libraries     []PipelineLibrary `json:"libraries,omitempty"`
+	Filters       []Filters         `json:"filters"`
+	Continuous    bool              `json:"continuous,omitempty"`
+}
+
+type PipelineState string
+
+const (
+	StatusDeploying  PipelineState = "DEPLOYING"
+	StatusStarting   PipelineState = "STARTING"
+	StatusRunning    PipelineState = "RUNNING"
+	StatusStopping   PipelineState = "STOPPPING"
+	StatusDeleted    PipelineState = "DELETED"
+	StatusRecovering PipelineState = "RECOVERING"
+	StatusFailed     PipelineState = "FAILED"
+	StatusResetting  PipelineState = "RESETTING"
+	StatusIdle       PipelineState = "IDLE"
+)
+
+type PipelineHealthStatus string
+
+const (
+	HealthStatusHealthy   PipelineHealthStatus = "HEALTHY"
+	HealthStatusUnhealthy PipelineHealthStatus = "UNHEALTHY"
+)
+
+type PipelineInfo struct {
+	PipelineId string                `json:"pipeline_id"`
+	Spec       *PipelineSpec         `json:"spec"`
+	State      *PipelineState        `json:"state"`
+	Cause      string                `json:"cause"`
+	ClusterId  string                `json:"cluster_id"`
+	Name       string                `json:"name"`
+	Health     *PipelineHealthStatus `json:"health"`
+}
+
+// Live Tables API Client
+
+type LiveTablesAPI struct {
+	client *common.DatabricksClient
+	ctx    context.Context
+}
+
+type LiveTableId string
```

Comment:
why do we need a type, if we don't overload it with additional methods? it results in overhead of casting it to string everywhere, without any added benefit.

Reply:
True. Wasn't sure how much we would benefit from type safety when I started, to prevent this API being called with another kind of ID. but now I see it is mainly self-contained, so can go back to plain string.